### PR TITLE
Fix load text property type

### PR DESCRIPTION
**Pull request by doctrino**
## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/data_classes/data_modeling/data_types.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None =
         type_ = resource["type"]
         obj: Any
         if type_ == "text":
-            obj = Text(is_list=resource["list"], collation=resource["collation"])
+            obj = Text(is_list=resource["list"], collation=resource.get("collation", "ucs_basic"))
```
Comment (nodegard) [inline cognite/client/data_classes/data_modeling/data_types.py:76]: **issue:** `is_list` should also be optional, and the same applies to all the other types. Also wouldn't it be best to not set the default value in the SDK and just rely on the API to set the default?

Comment (nodegard) [reply cognite/client/data_classes/data_modeling/data_types.py:76]: Nevermind "ucs_basic" is already defined as the default type for collation in `Text` so disregard the original comment but don't need to use that here

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,10 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [7.43.2] - 2024-05-10
+### Fixed
+- In containers, `PropertyType` `Text` required parameter `collation` is now optional, matching the API spec.
```
Comment (erlendvollset) [inline CHANGELOG.md:22]:
```suggestion
- In containers, `PropertyType` `Text` required parameter `collation` is now optional when `load()`ing, matching the API spec.
```

**Review by erlendvollset — APPROVED** [review]
LGTM

**Pull request merged**